ideascube/configuration/management/commands/config.py

```diff
@@ -0,0 +1,153 @@
+    get_all_namespaces, get_config_data, get_namespaced_configs,
+    )
+
+
+class Command(BaseCommand):
+    help = 'Manage server configuration'
+
+    def add_arguments(self, parser):
+        subs = parser.add_subparsers(
+            title='Commands', dest='cmd', metavar='',
+            parser_class=argparse.ArgumentParser)
+
+        get = subs.add_parser(
+            'get', help='Get the current value of a configuration option')
+        get.add_argument('namespace', help='The configuration namespace')
+        get.add_argument('key', help='The configuration key')
+        get.set_defaults(func=self.get_config)
+
+        help = subs.add_parser(
+            'help', help='Print help about a configuration option')
+        help.add_argument('namespace', help='The configuration namespace')
+        help.add_argument('key', help='The configuration key')
+        help.set_defaults(func=self.help_for_config)
+
+        list = subs.add_parser(
+            'list', help='List configuration namespaces and keys')
+        list.add_argument(
+            'namespace', nargs='?',
+            help='Only list configuration keys for this namespace')
+        list.set_defaults(func=self.list_configs)
+
+        reset = subs.add_parser(
+            'reset', help='Reset a configuration option to its default value')
+        reset.add_argument('namespace', help='The configuration namespace')
+        reset.add_argument('key', help='The configuration key')
+        reset.set_defaults(func=self.reset_config)
+
+        set = subs.add_parser(
+            'set', help='Set the value of a configuration option')
+        set.add_argument('namespace', help='The configuration namespace')
+        set.add_argument('key', help='The configuration key')
+        set.add_argument('value', help='The new value')
+        set.set_defaults(func=self.set_config)
+
+        self.parser = parser
+
+    def handle(self, *_, **options):
+        if 'func' not in options:
+            self.parser.print_help()
+            self.parser.exit(1)
+
+        options['func'](options)
+
+    def _evaluate_value(self, value):
+        if value in ('true', 'false'):
+            # Special case booleans for a better UX
+            value = value.capitalize()
+
+        try:
+            return ast.literal_eval(value)
+
+        except (SyntaxError, ValueError):
+            # Admin probably wanted a string and didn't use enough quotes
+            # examples:
+            #     $ ideascube config set server site-name thename
+            #     $ ideascube config set server site-name 'the name'
+            return str(value)
+
+    def get_config(self, options):
+        namespace = options['namespace']
+        key = options['key']
+
+        try:
+            value = get_config(namespace, key)
+
+        except (NoSuchConfigurationNamespaceError,
+                NoSuchConfigurationKeyError) as e:
+            raise CommandError(e)
+
+        print('%r' % value)
```

Comment:
> Maybe find a way to tell the user if the config is not set and we are returning the default value ?

I don't think it matters one bit for the user.

You want to know what languages are currently allowed for content uploads on the box. Whether it's the default value or not is irrelevant, all you want to know is the actual value.

Also, I totally expect the `get` command to be used very rarely (whether interactively or in scripts). The workflows I expect are:

* ansiblecube will use `set`
* an admin will sometimes want to `reset` if a value is causing problems on a box
* an admin will want to `get` the current value, and if it's not what they expected then `set` it to something else (or just `set` directly without even looking at the current value, because it often doesn't matter what the current value is when you're about to change it anyway)
* to do the above, an admin will need to `list` because they keep forgetting how that key is called, and then `help` to figure out what to put after `set`

Those are the primary workflows I designed the CLI for. Knowing whether the current value is the default isn't one of them (and it's possible anyway with `get` followed by `help`) because I don't expect it to ever be important.

Of course, actual use will test those assumptions, and sometimes prove them wrong. Again, this is why I'm interested in uninformed feedback from the main stakeholders here. (and I still won't mention them here to avoid influencing their testing)

We can always get back on this after such testing/feedback happened.

As for your actual suggestions, printing on stdout would break scripts as you said, so would printing on stderr (plus it would be bad taste). The only workable solution IMHO would be a `--verbose` or `--details` option. (and if we went there, I would think we could merge the functionality of `help` into that option of `get`)